catalog/core/catalog-core-standardframework/src/main/java/ddf/catalog/impl/operations/CreateOperations.java

```diff
@@ -117,96 +125,29 @@ public void setHistorian(Historian historian) {
-
-        validateCreateRequest(createRequest);
-
-        if (Requests.isLocal(createRequest)
-                && !sourceOperations.isSourceAvailable(sourceOperations.getCatalog())) {
-            SourceUnavailableException sourceUnavailableException = new SourceUnavailableException(
-                    "Local provider is not available, cannot perform create operation.");
-            if (INGEST_LOGGER.isWarnEnabled()) {
-                INGEST_LOGGER.warn(
-                        "Error on create operation, local provider not available. {} metacards failed to ingest. {}",
-                        createRequest.getMetacards()
-                                .size(),
-                        buildIngestLog(createRequest),
-                        sourceUnavailableException);
-            }
-            throw sourceUnavailableException;
-        }
-
         CreateResponse createResponse = null;
 
         Exception ingestError = null;
-        try {
-            createRequest = injectAttributes(createRequest);
-
-            setDefaultValues(createRequest);
-
-            Map<String, Serializable> unmodifiablePropertiesMap = Collections.unmodifiableMap(
-                    createRequest.getProperties());
-            HashMap<String, Set<String>> requestPolicyMap = new HashMap<>();
-            for (Metacard metacard : createRequest.getMetacards()) {
-                HashMap<String, Set<String>> itemPolicyMap = new HashMap<>();
-                for (PolicyPlugin plugin : frameworkProperties.getPolicyPlugins()) {
-                    PolicyResponse policyResponse = plugin.processPreCreate(metacard,
-                            unmodifiablePropertiesMap);
-                    opsSecuritySupport.buildPolicyMap(itemPolicyMap,
-                            policyResponse.itemPolicy()
-                                    .entrySet());
-                    opsSecuritySupport.buildPolicyMap(requestPolicyMap,
-                            policyResponse.operationPolicy()
-                                    .entrySet());
-                }
 
-                metacard.setAttribute(new AttributeImpl(Metacard.SECURITY, itemPolicyMap));
-            }
-            createRequest.getProperties()
-                    .put(PolicyPlugin.OPERATION_SECURITY, requestPolicyMap);
+        createRequest = queryOperations.setFlagsOnRequest(createRequest);
+        createRequest = validateCreateRequest(createRequest);
+        createRequest = validateLocalSource(createRequest);
 
-            for (AccessPlugin plugin : frameworkProperties.getAccessPlugins()) {
-                createRequest = plugin.processPreCreate(createRequest);
-            }
+        try {
+            createRequest = injectAttributes(createRequest);
+            createRequest = setDefaultValues(createRequest);
+            createRequest = updateCreateRequestPolicyMap(createRequest);
+            createRequest = processPrecreateAccessPlugins(createRequest);
```

Comment:
I agree. Couldn't we use the chain of responsibilities pattern and have the request go through a series of handlers instead of hard-coding the steps here? Or use decorators? I don't like the decorator approach as much as there would be a _lot_  of them which isn't as easy to follow, but I think it'd still be better than hard-coded, linear method calls.

And if our handlers are small, generic and somewhat configurable, we may be able to go as far as sharing them between different types of operations, e.g., create and update?